File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_map={}
        for i,num in enumerate(nums):
            compliment=target-num

            if compliment in num_map:
                return [num_map[compliment],i]
            
            num_map[num]=i

        return num_map

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int []ans=new int[2];
        ans[0]=ans[1]=-1;
        Map<Integer,Integer>mpp=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int num=nums[i];
            int moreneed=target-num;
            if(mpp.containsKey(moreneed)){
                ans[0]=mpp.get(moreneed);
                ans[1]=i;
                return ans;
            }
            mpp.put(nums[i],i);
        }
        return ans;
    }
}